src/app/qgsdxfexportdialog.cpp

```diff
@@ -106,24 +141,47 @@ int FieldSelectorDelegate::attributeIndex( const QAbstractItemModel *model, cons
 QgsVectorLayerAndAttributeModel::QgsVectorLayerAndAttributeModel( QgsLayerTree *rootNode, QObject *parent )
   : QgsLayerTreeModel( rootNode, parent )
 {
+  //init mCreateDDBlockInfo, mDDBlocksMaxNumberOfClasses
+  bool ddBlockDefaultEnabled = QgsSettings().value( QStringLiteral( "/qgis/dxfEnableDDBlocks" ), false ).toBool();
```

Comment:
this introduced a setting using the old API
see https://docs.qgis.org/3.34/en/docs/developers_guide/codingstandards.html#settings